pkg/workspace/controllers/metrics.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package controllers
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kaitov1beta1 "github.com/kaito-project/kaito/api/v1beta1"
+)
+
+var (
+	workspacePhaseCount = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "kaito_workspace_count",
+			Help: "Number of Workspaces in a certain phase (succeeded, error, pending, deleting)",
+		},
+		[]string{"phase"},
+	)
+)
+
+func monitorWorkspaces(ctx context.Context, k8sClient client.Client) {
+	ticker := time.NewTicker(1 * time.Minute)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			var wsList kaitov1beta1.WorkspaceList
+
+			if err := k8sClient.List(ctx, &wsList); err != nil {
+				klog.Errorf("failed to list all workspaces: %v", err)
+				continue
+			}
+
+			phaseCounts := map[string]float64{
+				"succeeded": 0,
+				"error":     0,
+				"pending":   0,
+				"deleting":  0,
+			}
+
+			for _, ws := range wsList.Items {
+				phase := determineWorkspacePhase(&ws)
+				if _, ok := phaseCounts[phase]; !ok {
+					phaseCounts[phase] = 0
+				}
+				phaseCounts[phase]++
+			}
+
```

Comment:
```suggestion
workspacePhaseCount.Reset()
```

Reply:
This will not happen. I think this code is right.

Reply:
This will not happen, it will be count from 0